pkl-commons-cli/src/main/kotlin/org/pkl/commons/cli/CliCommand.kt

```diff
@@ -152,9 +155,33 @@ abstract class CliCommand(protected val cliOptions: CliBaseOptions) {
     )
   }
 
-  // share HTTP client with other commands with the same cliOptions
-  protected val httpClient: HttpClient
-    get() = cliOptions.httpClient
+  /**
+   * The HTTP client used for this command.
+   *
+   * To release resources held by the HTTP client in a timely manner, call [HttpClient.close].
+   */
+  val httpClient: HttpClient by lazy {
+    with(HttpClient.builder()) {
+      setTestPort(cliOptions.testPort)
+      if (cliOptions.normalizedCaCertificates.isEmpty()) {
+        addDefaultCliCertificates()
+      } else {
+        for (file in cliOptions.normalizedCaCertificates) addCertificates(file)
+      }
+      if (cliOptions.proxyAddress != null) {
+        setProxy(cliOptions.proxyAddress, cliOptions.noProxy)
+      } else {
+        project?.evaluatorSettings?.http?.proxy?.let {
+          setProxy(it.address, it.noProxy ?: listOf())
+        }
+          ?: settings.http?.proxy?.let { setProxy(it.address, it.noProxy ?: listOf()) }
```

Comment:
I think this isn't what we want, right? I think we want the `address` and `noProxy` values each to be individually composed.

This way, this command means: use proxy settings from some lower precedence (~/.pkl/settings.pkl or system or whatever), except don't proxy localhost:

```
pkl eval --no-proxy localhost myModule.pkl
```

With the way it's written currently, it will throw away the `--no-proxy` flag.